src/credentials/CertificationDeclaration.cpp

```diff
@@ -473,12 +473,12 @@ CHIP_ERROR EncodeSignerInfo(const ByteSpan & signerKeyId, const P256ECDSASignatu
             }
             ASN1_END_SEQUENCE;
 
-            uint8_t asn1SignatureBuf[kMax_ECDSA_Signature_Length_Der];
-            MutableByteSpan asn1Signature(asn1SignatureBuf);
-            ReturnErrorOnFailure(EcdsaRawSignatureToAsn1(kP256_FE_Length, signature.Span(), asn1Signature));
-
             // signature OCTET STRING
-            ReturnErrorOnFailure(writer.PutOctetString(asn1Signature.data(), static_cast<uint16_t>(asn1Signature.size())));
+            ASN1_START_OCTET_STRING_ENCAPSULATED
+            {
+                ReturnErrorOnFailure(ConvertECDSASignatureRawToDER(P256ECDSASignatureSpan(signature.ConstBytes()), writer));
```

Comment:
As a followup: can we make EcdsaRawSignatureToAsn1 just share the implementation with ConvertECDSASignatureRawToDER?

@tcarmelveilleux do you happen to know why we have these two separate methods for doing the same thing?

Reply:
Yeah... it would also be good to figure out where these helpers "should" live... They don't seem strictly related to certificates, nor are they really part of the Cryto PAL (insofar as that PAL is actually abstraction away an underlying crypto backend). Maybe they should live in some CryptoUtils module.

Reply:
I'll resolve this for now, will follow up on Slack.